key_agreement.go

```diff
@@ -114,10 +114,18 @@ func md5SHA1Hash(slices [][]byte) []byte {
 // only used for >= TLS 1.2 and precisely identifies the hash function to use.
 func hashForServerKeyExchange(sigAndHash signatureAndHash, version uint16, slices ...[]byte) ([]byte, crypto.Hash, error) {
 	if version >= VersionTLS12 {
-		if !isSupportedSignatureAndHash(sigAndHash, supportedSignatureAlgorithms) {
+		var hashFunc crypto.Hash
+		var err error
+		if signatureSchemeIsPSS(sigAndHashToSigScheme(sigAndHash)) {
+			// 1.3 PSS schemes, when interpreted as 1.2
+			// Sig+Hash pairs, keep the hash in the
+			// signature byte.
```

Comment:
This is ugly as hell (but not your fault), as it's only a coincidence that the hash falls in the signature byte, the values should just be looked at as uint16. We might have to restructure how signatureAndHash works.